enumap.py

```diff
@@ -149,6 +148,34 @@ def _raise_invalid_args(cls, values, mapping, names):
                 f"missing keys {missing}; invalid keys {invalid}")
 
 
+def _type_cast_items(mapping, types):
+    """Generates key/value pairs for which each
+    value is casted with the callable in the `types` mapping.
+    """
+    key = value = None
+    try:
+        for key, type_callable in types.items():
+            yield key, type_callable(mapping[key])
+    except Exception as e:
```

Comment:
There's a (tiny) risk that this could mask unrelated exceptions like a `KeyError`. Could you be more specific about the exception types you catch, for example only `ValueError` (result of `int("2.0")` and such) and `TypeError`?

Reply:
That's a great thing to watch for in code like this, but we're actually safe courtesy of the call to `_make_checked_mapping`. That function makes sure we're working with good args/kwargs and raises exceptions as needed.

I kinda like the idea of only trapping `(Value|Type)Error`, but then weird callable type casters which raise `Exception` or `MySpecialException` wouldn't get the clear `TypeCastError`.